Add an AzureDevOps formatting style and warning suppression via regex

orj opened the pull request:
# 📝 Summary of Changes

Changes proposed in this pull request:

- Add an AzureDevOps formatting style
- In addition add the ability to suppress or elevate warnings to errors via a YML configuration file.
- Add support for logging out test failures.
- Add non-zero exit code flag.

# 🧐🗒 Reviewer Notes

## 💁 Example

Example output:
```
##vso[task.logissue type=warning;sourcepath=DSClients/Sources/DSClientMocks/MockClientFactory.swift;linenumber=11;columnnumber=16;]Stored property 'addressClient' of 'Sendable'-conforming class 'MockClientFactory' is mutable
##vso[task.logissue type=warning;sourcepath=DSComponents/Sources/DSComponents/Modifiers/StatusBarStyling.swift;linenumber=21;columnnumber=38;]'statusBarStyle' was deprecated in iOS 13.0: Use the statusBarManager property of the window scene instead.
##vso[task.logissue type=warning;sourcepath=DSComponents/Sources/DSComponents/Modifiers/StatusBarStyling.swift;linenumber=24;columnnumber=38;]'statusBarStyle' was deprecated in iOS 13.0: Use the statusBarManager property of the window scene instead.
##vso[task.logissue type=warning;sourcepath=Build/DerivedData/SourcePackages/checkouts/AppAuth-iOS/Source/AppAuth/iOS/OIDExternalUserAgentIOSCustomBrowser.m;linenumber=148;columnnumber=42;]'openURL:' is deprecated: first deprecated in iOS 10.0
##vso[task.logissue type=warning;sourcepath=Build/DerivedData/SourcePackages/checkouts/AppAuth-iOS/Source/AppAuth/iOS/OIDExternalUserAgentIOSCustomBrowser.m;linenumber=156;columnnumber=61;]'openURL:' is deprecated: first deprecated in iOS 10.0
##vso[task.logissue type=warning;sourcepath=Build/DerivedData/SourcePackages/checkouts/AppAuth-iOS/Source/AppAuth/iOS/OIDExternalUserAgentIOS.m;linenumber=180;columnnumber=58;]'openURL:' is deprecated: first deprecated in iOS 10.0
```

## 🔨 How To Test

```
xcrun xcresulttool get --format json --path Build.xcresult|swift run xcresult-issues -f azure-devops-logging -w warnings-example.yml
```

orj committed:
Add an AzureDevOps formatting style

In addition adds the ability to supress or elevate warnings to
errors via a YML configuration file.

orj committed:
Bump requirements to macOS 12 and Xcode 14.2

huwr reviewed (approved):
Seems good but I had trouble running locally, probably due to an existing issue unrelated to your change. Looks like `TestFailureIssueSummary` doesn't always have a `producingTarget` property (eg. in a UI test)

```
$ xcrun xcresulttool get --format json --path  /Users/huwr/Developer/Coles/DrinkStreet-iOS/Build/reports/Build/DerivedData/Logs/Test/Test-DrinkStreetUITests-2023.01.24_13-38-09-+1100.xcresult  | swift run xcresult-issues -f azure-devops-logging

Building for debugging...
Build complete! (0.19s)
Error: keyNotFound(CodingKeys(stringValue: "producingTarget", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "issues", intValue: nil), CodingKeys(stringValue: "testFailureSummaries", intValue: nil), CodingKeys(stringValue: "_values", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"producingTarget\", intValue: nil) (\"producingTarget\").", underlyingError: nil))
```

orj committed:
Correct lint issues

orj committed:
Add better handling of test failures

orj committed:
Add documentation

huwr commented on .swiftpm/xcode/xcshareddata/xcschemes/XCResultIssues.xcscheme:
```diff
@@ -79,16 +79,28 @@
             isEnabled = "YES">
          </CommandLineArgument>
+         <CommandLineArgument
+            argument = "/Users/oliver/Downloads/Test-DrinkStreetUITests-2023.01.24_13-38-09-+1100.xcresult.json"
```
Hrm… Not sure about this path

orj replied:
Whoops.

orj commented:
> Seems good but I had trouble running locally, probably due to an existing issue unrelated to your change. Looks like `TestFailureIssueSummary` doesn't always have a `producingTarget` property (eg. in a UI test)
> 
> ```
> $ xcrun xcresulttool get --format json --path  /Users/huwr/Developer/Coles/DrinkStreet-iOS/Build/reports/Build/DerivedData/Logs/Test/Test-DrinkStreetUITests-2023.01.24_13-38-09-+1100.xcresult  | swift run xcresult-issues -f azure-devops-logging
> 
> Building for debugging...
> Build complete! (0.19s)
> Error: keyNotFound(CodingKeys(stringValue: "producingTarget", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "issues", intValue: nil), CodingKeys(stringValue: "testFailureSummaries", intValue: nil), CodingKeys(stringValue: "_values", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"producingTarget\", intValue: nil) (\"producingTarget\").", underlyingError: nil))
> ```

Thanks @huwr.  I've addressed that issue and added logging of test failures to the tool.

huwr commented:
> > Seems good but I had trouble running locally, probably due to an existing issue unrelated to your change. Looks like `TestFailureIssueSummary` doesn't always have a `producingTarget` property (eg. in a UI test)
> > ```
> > $ xcrun xcresulttool get --format json --path  /Users/huwr/Developer/Coles/DrinkStreet-iOS/Build/reports/Build/DerivedData/Logs/Test/Test-DrinkStreetUITests-2023.01.24_13-38-09-+1100.xcresult  | swift run xcresult-issues -f azure-devops-logging
> > 
> > Building for debugging...
> > Build complete! (0.19s)
> > Error: keyNotFound(CodingKeys(stringValue: "producingTarget", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "issues", intValue: nil), CodingKeys(stringValue: "testFailureSummaries", intValue: nil), CodingKeys(stringValue: "_values", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"producingTarget\", intValue: nil) (\"producingTarget\").", underlyingError: nil))
> > ```
> 
> Thanks @huwr. I've addressed that issue and added logging of test failures to the tool.

Cool! That works heaps better now. Thanks.
